### PR TITLE
fix(win32): changes to allow working on windows

### PR DESCRIPTION
**Pull request by mikekidder**
Dude! Probably the most challenging submit I've had to date on Github.... Here you go.

-- no single-quotes in NPM package, escape with \"
-- **npm test cover** and ava bump .... reworked, wasnt getting output (no lines)
-- **context/index.test.js** .. absolute paths diff in windows
-- **entry/index.test.js** .. folder paths between systems ("/" vs "\")
-- **test/configs.js** ... couldn't find import

Reminder : remove the eslint rule for LF only

"npm run validate" run w/o errors. Please check

### CODE REVIEW

**Comment by kentcdodds** [comment]
Thanks! Looking now!

File: package.json [hunk 1]
```diff
@@ -10,10 +10,10 @@
     "prebuild": "rimraf dist",
     "validate": "npm-run-all --parallel lint cover build test:configs --sequential check-coverage",
     "build": "babel -d dist src",
-    "cover": "nyc --reporter=lcov --reporter=text --reporter=html npm run test",
+    "cover": "nyc npm test --reporter=lcov --reporter=text --reporter=html",
```
Comment (kentcdodds) [inline package.json]: Is this really necessary? It is counter-intuitive to put the command to cover before the options for the coverage...

File: package.json [hunk 2]
```diff
@@ -10,10 +10,10 @@
...
     "check-coverage": "nyc check-coverage --statements 100 --branches 100 --functions 100 --lines 100",
     "lint": "eslint .",
-    "test": "ava 'src/**/*.test.js' --verbose --require ./other/setup-ava-env.js",
+    "test": "ava \"src/**/*.test.js\" --verbose --require ./other/setup-ava-env.js",
```
Comment (kentcdodds) [inline package.json]: :+1:

File: package.json [hunk 3]
```diff
@@ -35,7 +35,7 @@
     "lodash": "4.4.0"
   },
   "devDependencies": {
-    "ava": "0.11.0",
+    "ava": "0.12.0",
```
Comment (kentcdodds) [inline package.json]: :+1:

File: src/validators/context/index.test.js [hunk 4]
```diff
@@ -3,7 +3,7 @@ import contextValidator from '.'
 const {validate} = contextValidator
 
 test('passes with an absolute path', t => {
-  const absolutePath = '/foo/bar/baz'
+  const absolutePath = process.platform === 'win32' ? 'C://foo/bar/baz' : '/foo/bar/baz'
```
Comment (kentcdodds) [inline src/validators/context/index.test.js]: :+1:

**Comment by kentcdodds** [comment]
Looks great. Just a few comments inline...

Also:

> Reminder : remove the eslint rule for LF only

Is there a way we could configure `.gitattributes` to force LF only? Like @BarryThePenguin [mentions](https://github.com/kentcdodds/webpack-validator/issues/7#issuecomment-189481247)

File: package.json (hunk 2 above)
Comment (BarryThePenguin) [inline package.json]: are quotes needed at all?

Comment (mikekidder) [inline package.json]: perhaps not, just knew single quotes dont fly in npm script via windows

Comment (BarryThePenguin) [inline package.json]: lets drop them

File: package.json (hunk 1 above)
Comment (mikekidder) [inline package.json]: I'll flip it back and re-check..

**Comment by mikekidder** [comment]
-- amend on commit, reverted **npm run cover** back, works for me!

**Comment by kentcdodds** [comment]
Do we need to say a .bit attributes file?

**Comment by mikekidder** [comment]
Still looking at it. Thought the whole CRLF vs LF was a non issue.  I've pushed many PRs with Mac users without issue, despite my Git autoconverting to CRLF

**Comment by BarryThePenguin** [comment]
My understanding of the problem with CRLF vs LF is that eslint is looking for LF. There's no problem with git converting between the two, that works just fine.

The problem is a regular windows user will checkout with CRLF locally and eslint flags that as an error

File: package.json [hunk 5]
```diff
@@ -13,7 +13,7 @@
     "cover": "nyc --reporter=lcov --reporter=text --reporter=html npm run test",
     "check-coverage": "nyc check-coverage --statements 100 --branches 100 --functions 100 --lines 100",
     "lint": "eslint .",
-    "test": "ava 'src/**/*.test.js' --verbose --require ./other/setup-ava-env.js",
+    "test": "ava \"src/**/*.test.js\" --verbose --require ./other/setup-ava-env.js",
```
Comment (BarryThePenguin) [inline package.json]: Dropping the `\"`? I don't think it's necessary

**Comment by mikekidder** [comment]
OK, so if you want to have Windows users to conform with LF only

**Windows Users** - be sure Git is not autoconverting repo files from LF to CRLF, otherwise linting will fail.

```git config --global core.autocrlf false```

However, like I said, first time I've run across this via Lint rules.
Don't think my PR converted the files.  Did it?

And if you're dealing with Visual Studio users who don't know the command line.  Good luck!

**Comment by kentcdodds** [comment]
You're right. I'll fix the linting rules. Thanks!

File: package.json (hunk 5 above)
Comment (mikekidder) [inline package.json]: fixed  --amended commit

**Comment by BarryThePenguin** [comment]
Changing the linting rule makes more sense than enforcing LF...

Next, Tabs vs Spaces! :100:

**Comment by kentcdodds** [comment]
Nope. Two spaces for life :-P

**Pull request merged**